src/components/input_fields/select_field.tsx

```diff
@@ -35,30 +35,41 @@ class SelectField extends React.Component<SelectFieldProps, SelectFieldState> {
     }
 
     private _onChange(value: string | number): void {
-        this.setState({touched: true});
-        this.props.onChangeHandler(this.props.id, value);
+        const { onChangeHandler, id } = this.props;
+        this.setState({ touched: true });
+        onChangeHandler(id, value);
     }
 
     private _staticRender(): JSX.Element | string {
-        const {value, selectOptions} = this.props;
-        const option = selectOptions.find(option => option.value === value);
+        const { value, selectOptions } = this.props;
+        const option = selectOptions.find(findOption => findOption.value === value);
         return option && option.label;
     }
 
     render() {
-        const {value, selectOptions, children, ...wrapperProps} = this.props;
-        const {id, staticField, validationError, submitted} = wrapperProps;
-        const {touched} = this.state;
-        return <InputFieldWrapper touched={touched} inputChildren={children} {...wrapperProps}>{
-            staticField ? this._staticRender() :
-            <SelectList
-                id={id}
-                onChangeHandler={this._onChange}
-                value={typeof value === 'number' || typeof value === 'string' ? value : null}
-                options={selectOptions}
-                error={validationError && (touched || submitted)}
-            />
-        }</InputFieldWrapper>;
+        const {
+            value, selectOptions, children, ...wrapperProps
+        } = this.props;
+        const {
+            id, staticField, validationError, submitted,
+        } = wrapperProps;
+        const { touched } = this.state;
+        return (
+            <InputFieldWrapper touched={touched} inputChildren={children} {...wrapperProps}>
+                {
+                    staticField ? this._staticRender()
```

Comment:
the previous version was much more readable..

Reply:
I've improved it